traits/trait_list_object.py

```diff
@@ -416,6 +416,23 @@ def rename(self, name):
             self.name = name
             self.trait = trait.handler
 
+    def clear(self):
+        if self.trait.minlen > 0:
+            self.len_error(0)
+
+        removed = list.copy(self)
+        list.clear(self)
+        index = 0
+
+        self._send_trait_items_event(
```

Comment:
As discussed in the meeting, in the case of an empty list there shouldn't be any items changed event. Please could you add a test for that corner case?

Reply:
I replaced the this with `del self[:]` but that causes a items changed notification when an empty list is cleared and thus, a test failure. Is this behavior for `del [:]` expected?

Reply:
Okay, that's interesting. :-) I'd expect `del mylist[:]` *not* to create an event if the list is already empty (and similarly, if `mylist` is a list of length 2, I'd expect something like `del mylist[2:]` not to create an event).

Let's keep this PR focused on `clear` and `copy`, but please could you open a bug report for the behaviour of `del` here?

Reply:
... and now I'm wondering what (for example) `mylist.extend([])` does. But again, that's off-topic for this PR.

Reply:
> please could you open a bug report for the behaviour of `del` here

Sorry, I forgot that you aren't around for the next couple of days; I've opened #738 for this.

Reply:
And for the record, `.extend([])` seems to be fine; it doesn't fire an event.